calculator.rb

```diff
@@ -0,0 +1,174 @@
+# Calculator
+# Platinum level
+# By Alice Rhomieux
+
+puts "Hello, welcome to the calculator."
+
+# List of supported operations, structured as a hash.
+# Each array is a family of terms that will be accepted for each operation.
+# The operator symbol is always the last element of the class so it can be selected with index [-1]
+# Oops, this is a global variable
+$supported_operations = {
+	addition: ["addition", "add", "plus", "+"],
+	subtraction: ["subtraction", "subtract", "minus", "sub", "-"],
+	multiplication: ["multiplication", "multiply", "times", "*"],
+	division: ["division", "divide", "div", "by", "/"],
+	exponent: ["exponent", "exp", "power of", "power", "**", "^"],
+	modulus: ["modulus", "modulo", "mod", "remainder", "%"]
+}
+
+# Check a number inputted by the user as a string and see if it has a decimal
+# If it has a decimal, convert it to a float, otherwise, convert it to an integer
+def check_num_format(input)
+	if input.include?(".")
+		data = input.delete(",").to_f
+	else
+ 		data = input.delete(",").to_i
+	end
+
+	return data
```

Comment:
The use of implicit returns is good, it's forcing you to be conscientious about what you want the function to return and how to structure your code to get to that result.

Once you're more comfortable with ruby, there's a little feature in ruby where functions will return the result of the last executed line of code. So you can do this:
```
def check_num_format(input)
  if input.include?(".")
    input.delete(",").to_f
  else
    input.delete(",").to_i
  end
end
```
Funky, huh?